src/xrpld/app/tx/detail/LoanSet.cpp

```diff
@@ -119,6 +119,10 @@ LoanSet::preflight(PreflightContext const& ctx)
             return *ret;
     }
 
+    if (auto const brokerID = ctx.tx[~sfLoanBrokerID];
+        brokerID && *brokerID == beast::zero)
+        return temINVALID;
```

Comment:
sfLoanBrokerID is required, so this doesn't have to be optional, but it's fine as-is.